plugins/modules/files/iso_customize.py

```diff
@@ -0,0 +1,335 @@
+        elif iso_type == "joliet":
+            record = opened_iso.get_record(joliet_path=dest_file)
+        elif iso_type == "udf":
+            record = opened_iso.get_record(udf_path=dest_file)
+
+        if record and not record.is_file():
+            module.fail_json(msg="The %s is not a file but directory." % dest_file)
+    except Exception as err:
+        msg = "Failed to get record of file %s in ISO with filesystem %s. error: %s" % (
+            dest_file, iso_type, to_native(err))
+        module.fail_json(msg=msg)
+
+    try:
+        if iso_type == "iso9660":
+            opened_iso.rm_hard_link(iso_path=file_in_iso_path)
+        elif iso_type == "rr":
+            opened_iso.rm_hard_link(iso_path=file_in_iso_path)
+        elif iso_type == "joliet":
+            opened_iso.rm_hard_link(joliet_path=dest_file)
+        elif iso_type == "udf":
+            # function "rm_hard_link" won't work in some OS (such as Ubuntu with python 3.10.4)
+            # take function "rm_file" instead
+            opened_iso.rm_file(udf_path=dest_file)
+    except Exception as err:
+        msg = "Failed to delete iso file %s with error: %s" % (dest_file, to_native(err))
+        module.fail_json(msg=msg)
+
+
+def iso_rebuild(module, src_iso, dest_iso, delete_files_list, add_files_list):
+    iso = None
+    iso_type = "iso9660"
+
+    try:
+        iso = pycdlib.PyCdlib()
+        iso.open(src_iso)
+        if iso.has_rock_ridge():
+            iso_type = "rr"
+        elif iso.has_joliet():
+            iso_type = "joliet"
+        elif iso.has_udf():
+            iso_type = "udf"
+
+        for item in delete_files_list:
+            iso_delete_file(module, iso, iso_type, item)
+
+        for item in add_files_list:
+            iso_add_file(module, iso, iso_type, item['src_file'], item['dest_file'])
+
+        iso.write(dest_iso)
+    except Exception as err:
+        msg = "Failed to rebuild ISO %s with error: %s" % (src_iso, to_native(err))
+        module.fail_json(msg=msg)
+    finally:
+        if iso:
+            iso.close()
+
+
+def main():
+    argument_spec = dict(
+        src_iso=dict(type='path', required=True),
+        dest_iso=dict(type='path', required=True),
+        delete_files=dict(type='list', elements='str', default=[]),
+        add_files=dict(
+            type='list', elements='dict', default=[],
+            options=dict(
+                src_file=dict(type='path', required=True),
+                dest_file=dict(type='str', required=True),
+            ),
+        ),
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        required_one_of=[('delete_files', 'add_files'), ],
+        supports_check_mode=True,
+    )
+    if not HAS_PYCDLIB:
+        module.fail_json(
+            missing_required_lib('pycdlib'), exception=PYCDLIB_IMP_ERR)
+
+    src_iso = module.params['src_iso']
+    if not os.path.exists(src_iso):
+        module.fail_json(msg="The %s does not exist." % src_iso)
```

Comment:
The wording needs a small adjustment:
```suggestion
    if not os.path.exists(src_iso):
        module.fail_json(msg="ISO file %s does not exist." % src_iso)
```
But maybe you wanna check for readability instead of just whether it exists. If you do that, then the message could be slightly adjusted to reflect that.